src/core/data.jl

```diff
@@ -1,23 +1,242 @@
+        for (comp_id, comp) in nw[comp_name]
+            if comp[status_key] !==_PMs.pm_component_status_inactive[comp_name]
+                active_count += 1
+            end
+        end
+    end
+        
+    return active_count
+end
+
+
 "Set damage status for damaged_items in nw_data"
-function damaged_items!(nw_data::Dict{String, Any}, damaged_items::Dict{String, Any})
+function damage_items!(nw_data::Dict{String,<:Any}, comp_list::Array{Tuple{String,String},1})
+    for (comp_name, comp_id) in comp_list
+        if haskey(nw_data, "multinetwork") && nw_data["multinetwork"] == true
+            for (nw, network) in nw_data["nw"]
+                network[comp_name][comp_id]["damaged"] = 1
+            end
+        else
+            nw_data[comp_name][comp_id]["damaged"] = 1
+        end
+    end
+end
 
-    for id in keys(damaged_items)
-        if haskey(nw_data, "nw")
-            for network in nw_data["nw"]
-                for i in damaged_items[id]
-                    nw_data[id][i]["damaged"] = 1
-                end
+
+""
+function get_repairable_items(network::Dict{String, Any})
+    if haskey(network, "multinetwork") && network["multinetwork"] == true
+        repairs = Dict{String,Any}("nw"=>Dict{String,Any}())
+        for (nw, net) in network["nw"]
+            repairs["nw"][nw] = _get_repairable_items(net)
+        end
+    else
+        repairs =  _get_repairable_items(network)
+    end
+    return repairs
+end
+
+
+""
+function _get_repairable_items(network::Dict{String,Any})
+    repairs = Array{Tuple{String,String},1}()
+    for (comp_name, status_key) in _PMs.pm_component_status
+        for (comp_id, comp) in get(network, comp_name, Dict())
+            if haskey(comp, status_key) && comp[status_key] != _PMs.pm_component_status_inactive[comp_name] && haskey(comp, "damaged") && comp["damaged"] == 1
+                push!(repairs, (comp_name, comp_id))
             end
+        end
+    end
+    return repairs
+end
+
+
+"Count the number of items with a key \"damaged\" == 1 in a network"
+function count_damaged_items(network::Dict{String, Any})
+    if _IMs.ismultinetwork(network)
+        Memento.warn(_PMs._LOGGER, "count_damaged_items supports single networks.  Attempting to select network 0.")
+        if haskey(network["nw"],"0")
+            Memento.info(_PMs._LOGGER, "Network 0 found.")
+            nw = network["nw"]["0"]
         else
-            for i in damaged_items[id]
-                nw_data[id][i]["damaged"] = 1
+            Memento.error(_PMs._LOGGER, "Network 0 not found.")
+        end
+    else
+        nw = network
+    end
+
+    damage_count = 0
+    for (comp_name, status_key) in _PMs.pm_component_status
+        for (i, comp) in get(nw, comp_name, Dict())
+            if haskey(comp, "damaged") && comp["damaged"] == 1
+                damage_count += 1
             end
         end
     end
+    
+    return damage_count
 end
 
+
+""
+function get_damaged_items(data::Dict{String,Any})
```

Comment:
This appears to be a duplicate functionality with `get/count_repairable_items`

Reply:
repairable is a subset of the damaged items that have an active status.